### PR TITLE
Fix return types for bcdiv() and bcmod() in PHP 8.0 and higher

### PR DESCRIPTION
**Pull request by W0rma**
Fixes https://github.com/phpstan/phpstan/issues/5134

### CODE REVIEW

File: src/Type/Php/BcMathStringOrNullReturnTypeExtension.php [hunk 1]
```diff
@@ -62,12 +80,30 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		$thirdArgument = $scope->getType($functionCall->getArgs()[2]->value);
-		$thirdArgumentIsNumeric = ($thirdArgument instanceof ConstantScalarType && is_numeric($thirdArgument->getValue())) || $thirdArgument instanceof IntegerType;
+		$thirdArgumentIsNumeric = false;
+		$thirdArgumentIsNegative = false;
+		if ($thirdArgument instanceof ConstantScalarType && is_numeric($thirdArgument->getValue())) {
+			$thirdArgumentIsNumeric = true;
+			$thirdArgumentIsNegative = ($thirdArgument->getValue() < 0);
+		} elseif ($thirdArgument instanceof IntegerType) {
```
Comment (staabm) [inline src/Type/Php/BcMathStringOrNullReturnTypeExtension.php]: I think this should use `IntegerType->isSuperTypeOf($thirdArgument)` instead

File: tests/PHPStan/Analyser/data/bcmath-dynamic-return-php8.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', $iNeg));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('0.00', 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt(0.0, 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('0', 1));
+\PHPStan\Testing\assertType('numeric-string|null', bcsqrt($iUnknown, $iUnknown));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', $iPos));
+\PHPStan\Testing\assertType('null', bcsqrt('-10', 0)); // Warning: Square root of negative number
+\PHPStan\Testing\assertType('null', bcsqrt($iNeg, 0));
+\PHPStan\Testing\assertType('null', bcsqrt('10', $nonNumeric)); // Warning: Second argument must be ?int (Fatal in PHP8)
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/bcmath-dynamic-return-php8.php]: when it fatals in php8, it should return `*NEVER*`

Comment (W0rma) [reply tests/PHPStan/Analyser/data/bcmath-dynamic-return-php8.php]: This PR was focused on `bcdiv()` and `bcmod()` only.

I just added fixes for `bcpowmod()` and `bcsqrt()`.

**Comment by ondrejmirtes** [comment]
Perfect, thank you.

**Pull request merged**